test/integration_test.py

```diff
@@ -30,399 +41,62 @@ def spack_spec(spec: str, log_filename: str = None):
-
-    def test_pytorch_fortran(self):
-        spack_spec('pytorch-fortran')
-
-    def test_pytorch_fortran_proxy(self):
-        spack_spec('pytorch-fortran-proxy')
-
-    def test_py_asttokens(self):
-        spack_spec('py-asttokens')
-
-    def test_py_black(self):
-        spack_spec('py-black')
-
-    def test_py_boltons(self):
-        spack_spec('py-boltons')
-
-    def test_py_cytoolz(self):
-        spack_spec('py-cytoolz')
-
-    def test_py_devtools(self):
-        spack_spec('py-devtools')
-
-    def test_py_editables(self):
-        spack_spec('py-editables')
-
-    def test_py_executing(self):
-        spack_spec('py-executing')
-
-    def test_py_factory_boy(self):
-        spack_spec('py-factory-boy')
-
-    def test_py_fprettify(self):
-        spack_spec('py-fprettify')
-
-    def test_py_frozendict(self):
-        spack_spec('py-frozendict')
-
-    def test_py_gridtools_cpp(self):
-        spack_spec('py-gridtools-cpp')
-
-    def test_py_gt4py(self):
-        spack_spec('py-gt4py')
-
-    def test_py_hatchling(self):
-        spack_spec('py-hatchling')
-
-    def test_py_icon4py(self):
-        spack_spec('py-icon4py')
+    log_with_spack(f'spack spec {spec}', 'integration_test', log_filename)
 
-    def test_py_inflection(self):
-        spack_spec('py-inflection')
 
-    def test_py_isort(self):
-        spack_spec('py-isort')
+@pytest.mark.parametrize('package', all_packages)
+def test_spack_info(package: str):
+    spack_info(package)
 
-    def test_py_lark(self):
-        spack_spec('py-lark')
 
-    def test_py_nanobind(self):
-        spack_spec('py-nanobind')
+@pytest.mark.parametrize('package', packages_for_machine())
+def test_spack_spec(package: str):
+    spack_spec(package)
 
-    def test_py_pathspec(self):
-        spack_spec('py-pathspec')
 
-    def test_py_pytest(self):
-        spack_spec('py-pytest')
+@pytest.mark.icon
+@pytest.mark.parametrize('variant', [
+    'serialization=create claw=std', 'fcgroup=DACE.externals/dace_icon.-O1',
+    'extra-config-args=--disable-new_feature,--enable-old_config_arg'
+])
+def test_icon_spec_with_variant(variant: str):
+    spack_spec(f'icon {variant}')
 
-    def test_py_pytest_factoryboy(self):
-        spack_spec('py-pytest-factoryboy')
 
-    def test_py_setuptools(self):
-        spack_spec('py-setuptools')
+#@pytest.mark.icon
+#@pytest.mark.parametrize('wrong_variant', [
+#    'fcgroup=dace.externals/dace_icon.-O1', 'extra-config-args=--new_feature'
+#])
+#@pytest.mark.xfail
+#def test_icon_spec_with_wrong_variant(wrong_variant: str):
+#    spack_spec(f'icon {wrong_variant}')
 
-    def test_py_sphinxcontrib_jquery(self):
-        spack_spec('py-sphinxcontrib-jquery')
 
-    def test_py_tabulate(self):
-        spack_spec('py-tabulate')
+@pytest.mark.cosmo
+@pytest.mark.parametrize(
+    'variant', ['cosmo_target=gpu ~cppdycore', 'cosmo_target=cpu ~cppdycore'])
+def test_cosmo_spec_with_variant(variant: str):
+    spack_spec(f'cosmo {variant}')
```

Comment:
Why does the cosmo test have a cosmo marker? That carries so little information.
![image](https://github.com/C2SM/spack-c2sm/assets/6570912/d6908c38-ead5-46aa-a502-2a24fb132119)

Reply:
Since we don't rely on parsing function/clasee-names anymore conditional tests need a marker.
I agree this is a bit too verbose, but at least it is bullet-proof.

Reply:
That's what review are for ;-)

An other option would be to create the marker on the fly based on the function name, itself prone to be forgotten or misspelled.

I went for that approach to also simplify the log-filename creation.